src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala

```diff
@@ -50,6 +53,16 @@ final case class OneStageEvaluationPlan(
 
   override def output: Seq[Attribute] = outputExpressions.map(_.toAttribute)
 
+  private val isProject = veFunction.functionName.startsWith("project")
+
+  private val copiedRefs = outputExpressions.filter(_.isInstanceOf[AttributeReference])
+  private val idsToCopy = child
+    .outputSet
+    .toList
+    .zipWithIndex
+    .collect {
+      case (ref, idx) if(copiedRefs.find(find => ref.name == find.name).isDefined) => idx
```

Comment:
should get unit tests here ,including some negative ones; i see this potentially filtering successfully although it shouldn't

Reply:
Sure, gonna write smth. But not really usre if there is any other way to compare those attrs.